core/Tracker/GoalManager.php

```diff
@@ -130,7 +130,7 @@ public function detectGoalsMatchingUrl($idSite, $action)
         $convertedGoals = array();
         foreach ($goals as $goal) {
             $convertedUrl = $this->detectGoalMatch($goal, $action);
-            if (!empty($convertedUrl)) {
+            if (!is_null($convertedUrl)) {
```

Comment:
`$convertedUrl` might be `false` if no 'url' is set (I think so by looking at the code but have not debugged it). Shouldn't still work though